src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,126 @@
+                "User cannot be null");
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("password123");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "Login must be at least 6 characters long");
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User user = new User();
+        user.setLogin("abc");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "Login must be at least 6 characters long");
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User user = new User();
+        user.setLogin("existingUser");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        when(storageDao.get("existingUser")).thenReturn(new User());
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "User with login existingUser already exists");
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword(null);
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "Password must be at least 6 characters long");
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("abc");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "Password must be at least 6 characters long");
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("password123");
+        user.setAge(null);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "User must be at least 18 years old");
+    }
+
+    @Test
+    void register_underageUser_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("password123");
+        user.setAge(17);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "User must be at least 18 years old");
+    }
+
+    @Test
+    void register_validUser_ok() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        when(storageDao.add(user)).thenReturn(user);
+
+        User result = registrationService.register(user);
+        assertEquals(user, result);
+        verify(storageDao, times(1)).add(user);
```

Comment:
The test 'register_validUser_ok' should also verify that the user has been added to the storage. Use 'verify(storageDao, times(1)).add(user);' to ensure that the user is correctly stored after registration.